Box.V2/Managers/BoxFoldersManager.cs

```diff
@@ -264,19 +264,32 @@ public BoxFoldersManager(IBoxConfig config, IBoxService service, IBoxConverter c
         /// before it was moved to the trash. If that parent folder no longer exists or if there is now an item with the same 
         /// name in that parent folder, the new parent folder and/or new name will need to be included in the request.
         /// </summary>
-        /// <returns></returns>
+        /// <param name="folderRequest">Folder request, which shall contain:
+        /// folderRequest.ID - The folder ID
+        /// folderRequest.Name - The new name for this item
+        /// folderRequest.Parent.Id - The id of the new parent folder
+        /// </param>
+        /// <param name="fields">Attribute(s) to include in the response</param>
+        /// <returns>The full item will be returned with a 201 Created status. By default it is restored to the parent folder it was in before it was trashed.</returns>
         public async Task<BoxFolder> RestoreTrashedFolderAsync(BoxFolderRequest folderRequest, List<string> fields = null)
         {
             folderRequest.ThrowIfNull("folderRequest")
                 .Id.ThrowIfNullOrWhiteSpace("folderRequest.Id");
+            folderRequest.Name.ThrowIfNullOrWhiteSpace("folderRequest.Name");
             folderRequest.Parent.ThrowIfNull("folderRequest.Parent")
                 .Id.ThrowIfNullOrWhiteSpace("folderRequest.Parent.Id");
 
             BoxRequest request = new BoxRequest(_config.FoldersEndpointUri, folderRequest.Id)
                     .Method(RequestMethod.Post)
-                    .Param(ParamFields, fields)
-                    .Payload(_converter.Serialize(folderRequest));
+                    .Param(ParamFields, fields);
+
+            // ID shall not be used in request body it is used only as url attribute
+            string oldId = folderRequest.Id;
```

Comment:
I decided to rememeber and renew ID value after processing to let it be transparent from caller point of view. I think it is not very good to change value of outside data internally may influence caller in some cases if he decides to use this value somehow again after method call.